src/entities/Piece.js

```diff
@@ -33,29 +38,35 @@ export class Piece {
         3 * CELL_HEIGHT,
         CELL_WIDTH,
         CELL_HEIGHT,
-        0x0000ff,
-        0.3
+        POSSIBLE_MOVES_GRID_COLOR,
+        0.2,
+        BOARD_BOARDER_COLOR
       );
       this.potentialMovesGrid.setInteractive();
       this.potentialMovesGrid.on("pointerdown", ({ x, y }) => {
         this.moveXY({ x, y });
-        this.unselect();
       });
       this.scene.sys.displayList.add(this.potentialMovesGrid);
+      this.scene.children.bringToTop(this.graphic);
     }
   };
 
-  unselect = () => {
+  moveXY = ({ x, y }) => {
+    // Board state update
```

Comment:
I'd like the piece to only move in the state and re-paint to handle any piece moves in the graphics world but I don't know if that's a good idea/pattern in this game making world.

The one place where that would help is Scarab swaps. Because otherwise we would need special code here to manage a Scarab swap graphically too.